### PR TITLE
plotly.js: Add modebar to layout in plotly js

### PR DESCRIPTION
**Pull request by JonathanMcIntyre**
Added modebar according to https://plotly.com/javascript/reference/#layout-modebar

### CODE REVIEW

**Comment by DangerBotOSS** [comment]
Inspecting the JavaScript source for this package found some properties that are not in the .d.ts files.
The check for missing properties isn't always right, so take this list as advice, not a requirement.
## plotly.js ([<kbd>unpkg</kbd>](https://unpkg.com/browse/plotly.js@latest/))
was missing the following properties:
1. version
1. Icons
1. Fx
1. Snapshot
1. PlotSchema
<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against 5238b2ba998cc56409dbf72ea31911838708b2eb
</p>

File: types/plotly.js/index.d.ts [hunk 1]
```diff
@@ -856,6 +857,16 @@ export interface Margin {
     pad: number;
 }
 
+export interface ModeBar {
+    activecolor: Color;
+    add: ModeBarDefaultButtons[];
```
Comment (peterblazejewicz) [inline types/plotly.js/index.d.ts:862]: docs say `ModeBarDefaultButtons | ModeBarDefaultButtons[]`:
```ts
Type: string or array of strings
```
```suggestion
    add: ModeBarDefaultButtons | ModeBarDefaultButtons[];
```

File: types/plotly.js/index.d.ts [hunk 2]
```diff
@@ -856,6 +857,16 @@ export interface Margin {
...
+    bgcolor: Color;
+    color: Color;
+    orientation: 'v' | 'h';
+    remove: ModeBarDefaultButtons[];
```
Comment (peterblazejewicz) [inline types/plotly.js/index.d.ts:866]: similar:
```suggestion
    remove: ModeBarDefaultButtons | ModeBarDefaultButtons[];
```
`Type: string or array of strings`

**Review by peterblazejewicz — APPROVED** [review]
LGTM!
@JonathanMcIntyre 
Next time please bring a test:
https://github.com/DefinitelyTyped/DefinitelyTyped#my-package-teststs

**Comment by JonathanMcIntyre** [comment]
Ready to merge

**Pull request merged**